src/Kiota.Builder/Writers/TypeScript/CodeFunctionWriter.cs

```diff
@@ -1,31 +1,57 @@
 ﻿
 
 using System;
-
+using System.Collections.Generic;
+using System.Linq;
 using Kiota.Builder.CodeDOM;
 using Kiota.Builder.Extensions;
 
 namespace Kiota.Builder.Writers.TypeScript;
 
 public class CodeFunctionWriter : BaseElementWriter<CodeFunction, TypeScriptConventionService>
 {
-    private readonly CodeUsingWriter _codeUsingWriter;
     public CodeFunctionWriter(TypeScriptConventionService conventionService, string clientNamespaceName) : base(conventionService)
     {
         _codeUsingWriter = new(clientNamespaceName);
     }
+    private TypeScriptConventionService? localConventions;
+    private readonly CodeUsingWriter _codeUsingWriter;
+    private static readonly HashSet<string> customSerializationWriters = new(StringComparer.OrdinalIgnoreCase) { "writeObjectValue", "writeCollectionOfObjectValues" };
 
     public override void WriteCodeElement(CodeFunction codeElement, LanguageWriter writer)
     {
         ArgumentNullException.ThrowIfNull(codeElement);
         if (codeElement.OriginalLocalMethod == null) throw new InvalidOperationException($"{nameof(codeElement.OriginalLocalMethod)} should not be null");
         ArgumentNullException.ThrowIfNull(writer);
         if (codeElement.Parent is not CodeNamespace) throw new InvalidOperationException("the parent of a function should be a namespace");
-
-        var returnType = conventions.GetTypeString(codeElement.OriginalLocalMethod.ReturnType, codeElement);
         _codeUsingWriter.WriteCodeElement(codeElement.StartBlock.Usings, codeElement.GetImmediateParentOfType<CodeNamespace>(), writer);
+        var codeMethod = codeElement.OriginalLocalMethod;
+
+        var returnType = codeMethod.Kind != CodeMethodKind.Factory ? conventions.GetTypeString(codeMethod.ReturnType, codeElement) : string.Empty;
         CodeMethodWriter.WriteMethodPrototypeInternal(codeElement.OriginalLocalMethod, writer, returnType, false, conventions, true);
+
         writer.IncreaseIndent();
+
+        localConventions = new TypeScriptConventionService(writer);
```

Comment:
I'm not sure if we should be re-initializing the conventions every time `WriteCodeElement` is called. This could potentially be unnecessarily expensive.

The base type also provides the `conventions` property that is initialized in the constructor. Any reason we can't replace `localConventions` with `conventions` from the base? 
(If this is because of the writer, I believe the `GetTypeString` function allows you to pass the writer as an optional param rather than creating a new Conventions instance every time)